ras_party/views/share_survey_view.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+import structlog
+from flask import Blueprint, request, make_response, jsonify, current_app
+from werkzeug.exceptions import BadRequest
+
+from ras_party.controllers import share_survey_controller
+from ras_party.controllers.business_controller import get_business_by_id
+from ras_party.controllers.validate import Validator, Exists
+from ras_party.views.account_view import auth
+
+share_survey_view = Blueprint('share_survey_view', __name__)
+logger = structlog.wrap_logger(logging.getLogger(__name__))
+
+
+@share_survey_view.before_request
+@auth.login_required
+def before_share_survey_view():
+    pass
+
+
+@auth.get_password
+def get_pw(username):
+    config_username = current_app.config['SECURITY_USER_NAME']
+    config_password = current_app.config['SECURITY_USER_PASSWORD']
+    if username == config_username:
+        return config_password
+
+
+@share_survey_view.route('/share-survey-users-count', methods=['GET'])
+def share_survey_users():
+    """
+    Get total users count who are already enrolled and pending share survey against business id and survey id
+    """
+    payload = request.get_json()
+    try:
+        business_id = payload['business_id']
+        survey_id = payload['survey_id']
+        # this is just to validate that the business_id exists
+        get_business_by_id(business_id)
```

Comment:
The responses for this endpoint aren't quite right either I think.  I'd expect a 400 if either wasn't supplied.  A 404 if the business/survey doesn't exist and a 200 on success.  It's more of a faff, but it also allows for more sensible errors to be given to the clients of this API

Reply:
business/survey returns bad request. will include the request parameter 400 error